core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1467,6 +1466,83 @@ object GroupMetadataManager {
     }
   }
 
+  /**
+   * Exposed for printing records using [[kafka.tools.DumpLogSegments]]
+   */
+  def formatRecordKeyAndValue(record: Record): (Option[String], Option[String]) = {
```

Comment:
Can one or the other string be missing? Or will they always be present or absent together?

Reply:
I agree the API is a bit awkward, but I think both cases are possible. For example, a tombstone will have a key, but not a value.